### ISSUE TITLE
[bug] <title> Problem running config.after callback: Cannot read properties of undefined (reading 'enabledMiddleware')

### ISSUE THREAD

**Issue by AlexShRz** [roots/bud#2664]
### Agreement

- [X] This is not a duplicate of an existing issue
- [X] I have read the [guidelines for Contributing to Roots Projects](https://github.com/roots/.github/blob/master/CONTRIBUTING.md)
- [X] This is not a personal support request that should be posted on the [Roots Discourse](https://discourse.roots.io/) community

### Describe the issue

👉 **What I'm trying to do:**
1. I'm trying to setup multiple compilation instances to use different tailwind configs - one for the block editor, and another one for the app itself (what we render using blade templates).
2. `pnpm run build` works fine, but `pnpm run dev` throws an error.

📜 **Project info**
1. I'm using NX monorepo and pnpm to manage my dependencies.
2. Using Sage Wp with wp-env (no bedrock).

👉 **What I've already tried**
1. Running `bud clean` - no results.
2. Deleting all node_modules, pnpm.lock and running `pnpm store prune && pnpm install --force --public-hoist-pattern=*` in monorepo root - no results.
 

💻 **Machine info**
OS: Mac os, m1

**Package JSON**
```
{
  "name": "[name]",
  "private": true,
  "$schema": "https://bud.js.org/bud.package.json",
  "browserslist": [
    "extends @roots/browserslist-config"
  ],
  "engines": {
    "node": ">=20.0.0"
  },
  "type": "module",
  "scripts": {
    "dev": "bud dev",
    "build": "bud build",
    "translate": "npm run translate:pot && npm run translate:update",
    "translate:pot": "wp i18n make-pot . ./resources/lang/sage.pot --include=\"theme.json,patterns,app,resources\"",
    "translate:update": "for file in ./resources/lang/*.po; do wp i18n update-po ./resources/lang/sage.pot $file; done",
    "translate:compile": "npm run translate:mo && npm run translate:js",
    "translate:js": "wp i18n make-json ./resources/lang --pretty-print",
    "translate:mo": "wp i18n make-mo ./resources/lang ./resources/lang",
  },
  "devDependencies": {
    "@roots/bud": "^6.23.3",
    "@roots/bud-tailwindcss": "^6.23.3",
    "@roots/sage": "^6.23.3",
    "@svgr/webpack": "^6.4.0",
    "ts-loader": "^9.5.1",
    "vitest": "1.6.0",
    "webpack-node-externals": "^3.0.0"
  },
  "dependencies": {
    "class-variance-authority": "^0.7.1",
    "clsx": "^2.1.1",
    "lucide-react": "^0.461.0",
    "tailwind-merge": "^2.5.5",
    "tailwindcss-animate": "^1.0.7",
    "yaml-loader": "^0.8.0"
  }
}
```

**ts.config.json**
```
{
  "extends": [
    "../../tsconfig.base.json",
    "@roots/sage/config/tsconfig.json"
  ],
  "compilerOptions": {
    "module": "CommonJS",
    "moduleResolution": "Bundler",
    "forceConsistentCasingInFileNames": true,
    "strict": true,
    "allowImportingTsExtensions": true,
    "isolatedModules": true,
    "noImplicitOverride": true,
    "noPropertyAccessFromIndexSignature": true,
    "noImplicitReturns": true,
    "noFallthroughCasesInSwitch": true,
    "paths": {
      "@/*": [
        "apps/[appName]-toolkit/block-editor/src/*"
      ],
      "@server-components/*": [
        "apps/[appName]-toolkit/app/View/Components/*"
      ],
      "@server-blocks/*": [
        "apps/[appName]-toolkit/app/View/Blocks/*"
      ],
      "@[appName]/api-test-utils": [
        "libs/api-test-utils/src/index.ts"
      ],
      "@[appName]/background-task-worker": [
        "libs/background-task-worker/src/index.ts"
      ],
      "@[appName]/blocks": [
        "libs/blocks/src/index.ts"
      ],
      "@[appName]/command-bus": [
        "libs/command-bus/src/index.ts"
      ],
      "@[appName]/core-api": [
        "libs/core-api/src/index.ts"
      ],
      "@[appName]/core-api/test-utils": [
        "libs/core-api/src/testUtils.ts"
      ],
      "@[appName]/post-commands": [
        "libs/post-commands/src/index.ts"
      ],
      "@[appName]/post-content-tree": [
        "libs/post-content-tree/src/index.ts"
      ],
      "@[appName]/post-content-tree/test-utils": [
        "libs/post-content-tree/src/testUtils.ts"
      ],
      "@[appName]/theme-ui": [
        "libs/theme-ui/src/index.ts"
      ],
      "@[appName]/type-utils": [
        "libs/type-utils/src/index.ts"
      ]
    }
  },
  "files": [
    "bud.config.ts"
  ],
  "include": [
    "resources",
    "../../libs"
  ],
  "exclude": [
    "node_modules",
    "public"
  ],
  "references": [
    {
      "path": "./tsconfig.lib.json"
    },
    {
      "path": "./tsconfig.spec.json"
    }
  ]
}
```

**tsconfig.lib.json**
```
{
  "extends": "./tsconfig.json",
  "compilerOptions": {
    "outDir": "../../dist/out-tsc",
    "declaration": true,
    "types": [
      "node"
    ],
  },
  "include": [
    "**/*.ts",
    "*.ts"
  ],
  "exclude": [
    "/**/*.spec.ts",
    "/**/*.test.ts"
  ]
}
```

**tsconfig.spec.json**
```
{
	"extends": "./tsconfig.json",
	"compilerOptions": {
		"outDir": "../../dist/out-tsc",
		"types": [
			"node"
		]
	},
	"include": [
		"./vitest.config.ts",
		"src/**/*.test.ts",
		"src/**/*.spec.ts",
		"src/**/*.test.tsx",
		"src/**/*.spec.tsx",
		"src/**/*.test.js",
		"src/**/*.spec.js",
		"src/**/*.test.jsx",
		"src/**/*.spec.jsx",
		"src/**/*.d.ts"
	]
}
```

**tsconfig.base.json** (monorepo root)
```
{
  "compileOnSave": false,
  "compilerOptions": {
    "rootDir": ".",
    "sourceMap": true,
    "declaration": false,
    "moduleResolution": "node",
    "forceConsistentCasingInFileNames": true,
    "esModuleInterop": true,
    "noImplicitOverride": true,
    "noPropertyAccessFromIndexSignature": false,
    "noImplicitReturns": true,
    "noFallthroughCasesInSwitch": true,
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "resolveJsonModule": true,
    "importHelpers": true,
    "target": "es2015",
    "module": "esnext",
    "lib": [
      "es2020",
      "dom"
    ],
    "baseUrl": ".",
    "skipLibCheck": true,
    "skipDefaultLibCheck": true,
    "noEmit": false,
    "noEmitOnError": false,
    "paths": {
      "@[appName]/api-test-utils": [
        "libs/api-test-utils/src/index.ts"
      ],
      "@[appName]/background-task-worker": [
        "libs/background-task-worker/src/index.ts"
      ],
      "@[appName]/blocks": [
        "libs/blocks/src/index.ts"
      ],
      "@[appName]/command-bus": [
        "libs/command-bus/src/index.ts"
      ],
      "@[appName]/core-api": [
        "libs/core-api/src/index.ts"
      ],
      "@[appName]/core-api/test-utils": [
        "libs/core-api/src/testUtils.ts"
      ],
      "@[appName]/post-commands": [
        "libs/post-commands/src/index.ts"
      ],
      "@[appName]/post-content-tree": [
        "libs/post-content-tree/src/index.ts"
      ],
      "@[appName]/post-content-tree/test-utils": [
        "libs/post-content-tree/src/testUtils.ts"
      ],
      "@[appName]/theme-ui": [
        "libs/theme-ui/src/index.ts"
      ],
      "@[appName]/type-utils": [
        "libs/type-utils/src/index.ts"
      ]
    },
    "jsx": "react"
  },
  "exclude": [
    "node_modules",
    "tmp",
    "build",
    ".idea",
    "__tests__",
    "vendor"
  ]
}
```
**package.json (monorepo root)**
```
{
  "name": "@[appName]",
  "version": "0.0.0",
  "license": "MIT",
  "scripts": {
    "i": "pnpm store prune && pnpm install --force --public-hoist-pattern=*",
    "start-wp": "wp-env start --xdebug",
    "clean-wp": "wp-env clean all",
    "restart-wp": "pnpm run clean-wp && pnpm run start-wp",
  },
  "private": true,
  "devDependencies": {
    "@nx/cypress": "19.4.3",
    "@nx/devkit": "19.4.3",
    "@nx/eslint": "19.4.3",
    "@nx/eslint-plugin": "19.4.3",
    "@nx/jest": "19.4.3",
    "@nx/js": "19.4.3",
    "@nx/node": "19.4.3",
    "@nx/playwright": "19.4.3",
    "@nx/react": "19.4.3",
    "@nx/vite": "19.4.3",
    "@nx/web": "19.4.3",
    "@nx/webpack": "19.4.3",
    "@nx/workspace": "19.4.3",
    "@playwright/test": "^1.36.0",
    "@rollup/plugin-typescript": "^11.1.6",
    "@swc-node/register": "1.9.2",
    "@swc/cli": "0.3.14",
    "@swc/core": "1.5.7",
    "@swc/helpers": "~0.5.11",
    "@swc/jest": "~0.2.36",
    "@testing-library/react": "15.0.6",
    "@types/jest": "^29.4.0",
    "@types/node": "18.19.9",
    "@types/react": "18.3.1",
    "@types/react-dom": "18.3.0",
    "@typescript-eslint/eslint-plugin": "7.9.0",
    "@typescript-eslint/parser": "7.9.0",
    "@vitejs/plugin-react": "^4.2.0",
    "@vitest/coverage-v8": "1.6.0",
    "@vitest/ui": "1.6.0",
    "babel-jest": "^29.4.1",
    "eslint": "8.57.0",
    "eslint-config-prettier": "^9.0.0",
    "eslint-plugin-import": "2.27.5",
    "eslint-plugin-jsx-a11y": "6.7.1",
    "eslint-plugin-playwright": "^0.15.3",
    "eslint-plugin-react": "7.32.2",
    "eslint-plugin-react-hooks": "4.6.0",
    "jest": "^29.4.1",
    "jest-environment-jsdom": "^29.4.1",
    "jest-environment-node": "^29.4.1",
    "jsdom": "~22.1.0",
    "nx": "19.4.3",
    "postcss": "8.4.38",
    "postcss-calc": "^8.2.4",
    "postcss-cli": "^8.2.0",
    "postcss-css-variables": "^0.18.0",
    "postcss-custom-media": "^8.0.2",
    "postcss-custom-properties": "^13.1.4",
    "postcss-discard-duplicates": "^5.1.0",
    "postcss-focus-within": "^6.1.1",
    "postcss-for": "^2.1.1",
    "postcss-import": "^15.1.0",
    "postcss-merge-rules": "^5.1.2",
    "postcss-nested": "^5.0.6",
    "prettier": "^2.6.2",
    "rollup-plugin-typescript-paths": "^1.5.0",
    "swc-loader": "0.1.15",
    "ts-jest": "^29.1.0",
    "ts-node": "10.9.1",
    "tslib": "^2.6.2",
    "typescript": "5.4.5",
    "vite": "^5.0.0",
    "vite-plugin-dts": "~3.8.1",
    "vite-tsconfig-paths": "^4.3.1",
    "vitest": "1.6.0",
    "webpack-cli": "^5.1.4",
    "zip-local": "^0.3.5"
  },
  "dependencies": {
    "react": "18.3.1",
    "react-dom": "18.3.1"
  }
}
```

### Expected Behavior

I expect my dev server to run without any issues.

### Actual Behavior

I get this error:
```
pnpm run dev

> [PROJECT]@ dev /Users/[USER]/Desktop/[MONOREPO]/apps/[PROJECT]
> bud dev

BudError

✘ problem running config.after callback: Cannot read properties of undefined (reading 'enabledMiddleware')

ℹ Stack trace

 at /Users/[USER]/Desktop/[MONOREPO]/node_modules/.pnpm/ink@4.4.1_@types+react@18.3.1_react@18.3.1/node_modules/ink/build/components/Static.js:24:16
 at Array.map (<anonymous>)
 at Static (/Users/[USER]/Desktop/[MONOREPO]/node_modules/.pnpm/ink@4.4.1_@types+react@18.3.1_react@18.3.1/node_modules/ink/build/components/Static.js:23:36)
 at renderWithHooks (/Users/[USER]/Desktop/[MONOREPO]/node_modules/.pnpm/react-reconciler@0.29.2_react@18.3.1/node_modules/react-reconciler/cjs/react-reconciler.development.js:6659:18)
 at mountIndeterminateComponent (/Users/[USER]/Desktop/[MONOREPO]/node_modules/.pnpm/react-reconciler@0.29.2_react@18.3.1/node_modules/react-reconciler/cjs/react-reconciler.development.js:11276:13)
 at beginWork (/Users/[USER]/Desktop/[MONOREPO]/node_modules/.pnpm/react-reconciler@0.29.2_react@18.3.1/node_modules/react-reconciler/cjs/react-reconciler.development.js:12799:16)
 at beginWork$1 (/Users/[USER]/Desktop/[MONOREPO]/node_modules/.pnpm/react-reconciler@0.29.2_react@18.3.1/node_modules/react-reconciler/cjs/react-reconciler.development.js:19608:14)
 at performUnitOfWork (/Users/[USER]/Desktop/[MONOREPO]/node_modules/.pnpm/react-reconciler@0.29.2_react@18.3.1/node_modules/react-reconciler/cjs/react-reconciler.development.js:18742:12)


```

### Steps To Reproduce

Unfortunately, I'm not quite sure what is happening, maybe something with bud cache or how pnpm works? 

👉 **Info that might be useful**
1. This problem happens only when using bud.make(), normal configs work fine
2. The issue seems to be related to the `@roots/bud-react/lib/react-refresh` package, it fails here because `bud.server` is undefined:
```
let BudReactRefresh = class BudReactRefresh extends Extension {
    /**
     * {@link Extension.configAfter}
     */
    async configAfter(bud) {
        if (!bud.isDevelopment)
            return;
            /// 👉 FAILS HERE, bud.server is undefined
        if (!(`hot` in bud.server.enabledMiddleware))
            return;
```
3. I've tried putting `console.log` inside `configAfter` - it seems to be running twice (maybe for each config inside `bud.make()`?). When my console.log gets called for the first time, bud.server is not undefined, but during the second run it is.

### version

6.23.3

### Configuration

```zsh
[bud.config.ts]

import type { Bud } from '@roots/bud';
// import path from 'node:path';
import getThemeBudConfig from './themeConfig.ts';
import getBlockEditorBudConfig from './blockEditorConfig.ts';

// const baseJsonConfig: {
//     compilerOptions: {
//         paths: Record<string, string[]>;
//     };
// } = require('../../tsconfig.base.json');
//
// const buildableLibraryPaths = Object.entries(baseJsonConfig.compilerOptions.paths).reduce<Record<string, string>>(
//     (final, [libName, paths]) => ({
//         ...final,
//         [libName]: path.resolve(__dirname, '../..', paths[0]),
//     }),
//     {}
// );

export default async (bud: Bud) => {
    // Set up buildable nx library paths
    bud
        // .setPath(buildableLibraryPaths)
        // .compilePaths(Object.keys(buildableLibraryPaths).map((path) => bud.path(path)))
        .setUrl('http://localhost:3000')
        .setProxyUrl('http://localhost:8888')
        .when(bud.isDevelopment, (bud) => bud.devtool('source-map'))
        .watch(['app', 'resources/views', 'resources', 'resources/styles', 'routes', 'block-editor/src'])
        .tap((bud) => {
            /*
             * Make sure css files are standalone in dev mode and not merged
             * with js files.
             * */
            bud.hooks.action('build.before', ((bud: Bud) => {
                bud.extensions.get('@roots/bud-extensions/mini-css-extract-plugin').enable(true);
            }) as any);
            bud.build.items.precss.setLoader('minicss');
        })
        .tap((bud) => {
            /*
             * Setup svg icons and svgo that should be accessible for both
             * block editor and server
             * */
            bud.hooks.on('build.module.rules.oneOf', (rules = []) => {
                rules.unshift({
                    test: /\.svg$/,
                    use: ['@svgr/webpack'],
                });

                return rules;
            });
        })
        .alias('@icons', bud.path('resources/images/icons'));

    // Setup individual configs for block editor controls and the theme
    await bud.make({ label: 'theme', basedir: bud.path('./') }, getThemeBudConfig as any);
    await bud.make({ label: 'block-editor', basedir: bud.path('./block-editor') }, getBlockEditorBudConfig as any);
};


--------------------------------------------------------
[blockEditorConfig.ts]

import { Bud } from '@roots/bud';
import BudTailwindCss from '@roots/bud-tailwindcss';

export default (bud: Bud) => {
    return bud
        .setPath('@src', bud.path(''))
        .setPublicPath('/public')
        .entry('block-editor', [bud.path('src/index.ts'), bud.path('src/index.css')])
        .tap((bud) => {
            (bud.tailwind as BudTailwindCss).setConfigPath(bud.path('tailwind-editor.config.js'));
        })
        .tap((bud) => {
            /*
             * Add yaml loader to load shared tailwind classes
             * */
            bud.hooks.on('build.module.rules.oneOf', (rules) => [
                ...(rules || []),
                {
                    test: /\.ya?ml$/,
                    use: 'yaml-loader',
                },
            ]);
        })
        .tap((bud) => {
            /*
             * Set unique cache name
             * */
            bud.cache.name = 'block-editor';
        });
};
--------------------------------------------------------
[themeConfig.ts]

import { Bud } from '@roots/bud';

export default (bud: Bud) => {
    return bud
        .entry('app', ['@scripts/app', '@styles/app'])
        .entry('editor', ['@scripts/editor', '@styles/editor'])
        .assets(['images'])
        .setPublicPath('/public')
        .copyDir(['fonts', 'fonts'])
        .wpjson.setSettings({
            background: {
                backgroundImage: true,
            },
            color: {
                custom: false,
                customDuotone: false,
                customGradient: false,
                defaultDuotone: false,
                defaultGradients: false,
                defaultPalette: false,
                duotone: [],
            },
            custom: {
                spacing: {},
                typography: {
                    'font-size': {},
                    'line-height': {},
                },
            },
            spacing: {
                padding: true,
                units: ['px'],
            },
            typography: {
                customFontSize: false,
            },
        })
        .useTailwindColors()
        .useTailwindFontFamily()
        .useTailwindFontSize();
};
```

### Relevant .budfiles

```zsh
pnpm run dev --debug

> dev
> bud dev


 BudError 

✘ problem running config.after callback: Cannot read properties of undefined (reading 'enabledMiddleware')

ℹ Stack trace

 at ./node_modules/ink/build/components/Static.js:24:16
 at Array.map (<anonymous>)
 at Static (./node_modules/ink/build/components/Static.js:23:36)
 at renderWithHooks (./node_modules/react-reconciler/cjs/react-reconciler.development.js:6659:18)
 at mountIndeterminateComponent (./node_modules/react-reconciler/cjs/react-reconciler.development.js:11276:13)
 at beginWork (./node_modules/react-reconciler/cjs/react-reconciler.development.js:12799:16)
 at beginWork$1 (./node_modules/react-reconciler/cjs/react-reconciler.development.js:19608:14)
 at performUnitOfWork (./node_modules/react-reconciler/cjs/react-reconciler.development.js:18742:12)
```

**Comment by kellymears** [comment]
This came up in the multi-compiler config we use at work as well. 

It _should_ be fixed in 6.24.0.

**Comment by AlexShRz** [comment]
Can confirm that it works 👍. Thanks man
